piker/brokers/binance.py

```diff
@@ -492,5 +545,32 @@ async def stream_quotes(
             # start streaming
             async for typ, msg in msg_gen:
 
-                topic = msg['symbol']
+                topic = msg['symbol'].lower()
                 await send_chan.send({topic: msg})
+
+
+@tractor.context
+async def open_symbol_search(
+    ctx: tractor.Context,
+) -> Client:
+    async with open_cached_client('binance') as client:
+
+        # load all symbols locally for fast search
+        cache = await client.cache_symbols()
+        await ctx.started()
+
+        async with ctx.open_stream() as stream:
+
+            async for pattern in stream:
+                # results = await client.symbol_info(sym=pattern.upper())
```

Comment:
Yeah turns out they don't have a "search" in the API they offer, you have to have the exact symbol.

Collecting the whole cache at client startup is reasonably fast anyway (and we do it async with the other backends), so all gud i think.